examples/pytorch/ogb/seal_ogbl/utils.py

```diff
@@ -0,0 +1,132 @@
+import sys
+import random
+import numpy as np
+from scipy.sparse.csgraph import shortest_path
+import torch
+from dgl.sampling import global_uniform_negative_sampling
+
+
+def k_hop_subgraph(src, dst, num_hops, g, sample_ratio=1., directed=False):
+    # Extract the k-hop enclosing subgraph around link (src, dst) from g
+    nodes = [src, dst]
+    visited = set([src, dst])
+    fringe = set([src, dst])
```

Comment:
I don't quite like using Python's builtin `set` and `list`.  Consider using `numpy.intersect1d` and stuff for set functions.